internal/backend/s3/s3.go

```diff
@@ -87,6 +87,53 @@ func open(ctx context.Context, cfg Config, rt http.RoundTripper) (*Backend, erro
 		debug.Log("using anonymous access for %#v", cfg.Endpoint)
 	}
 
+	// assume role
+	roleArn := os.Getenv("AWS_ASSUME_ROLE_ARN")
```

Comment:
Please extract the code into a helper function. The `open()` function is already far too long.

Reply:
Done